src/Symfony/Component/Ldap/Adapter/AbstractConnection.php

```diff
@@ -24,22 +24,51 @@
     public function __construct(array $config = array())
     {
         $resolver = new OptionsResolver();
+
+        $this->configureOptions($resolver);
+
+        $this->config = $resolver->resolve($config);
+    }
+
+    /**
+     * Configures the adapter's options.
+     *
+     * @param OptionsResolver $resolver An OptionsResolver instance
+     */
+    protected function configureOptions(OptionsResolver $resolver)
+    {
         $resolver->setDefaults(array(
-            'host' => null,
+            'host' => 'localhost',
             'port' => 389,
             'version' => 3,
-            'useSsl' => false,
-            'useStartTls' => false,
-            'optReferrals' => false,
+            'connectionString' => null,
+            'ssl' => false,
+            'tls' => false,
+            'options' => array(),
         ));
-        $resolver->setNormalizer('host', function (Options $options, $value) {
-            if ($value && $options['useSsl']) {
-                return 'ldaps://'.$value;
-            }
 
-            return $value;
+        $resolver->setAllowedTypes('host', 'string');
+        $resolver->setAllowedTypes('port', 'numeric');
+        $resolver->setAllowedTypes('version', 'numeric');
+        $resolver->setAllowedTypes('ssl', 'bool');
+        $resolver->setAllowedTypes('tls', 'bool');
+        $resolver->setAllowedTypes('options', 'array');
+
+        $resolver->setNormalizer('port', function (Options $options, $value) {
```

Comment:
rather than using a normalizer to change the port for SSL, it would be better to use a lazy option as default value for the port, making the default value dependant on the ``ssl`` option